### ISSUE TITLE
Waiting for results... or Submission timed out but Status is "All Systems Operational"

### ISSUE THREAD

**Issue by tehnicaorg** [Codewars/codewars.com#335]
> Submission timed out while communicating with our servers. This may infrequently happen if our servers are extremly busy. Please try again. If issues persist you can check status.codewars.com to see if server issues are being investigated.

https://codewars.statuspage.io/ shouldn't say "All systems operational" if multiple timeouts happen.

I've tried http://www.codewars.com/kata/54d4c8b08776e4ad92000835/train/java even with a function that just returns a null, without doing any calculations.

**Comment by redscaresu** [comment]
Same for me :(

**Comment by jhoffner** [comment]
@redscaresu are you also having issues with Java or another language? Based off of my server logs it looks like only Java may be affected

**Comment by redscaresu** [comment]
hey thanks for looking into this, no I am using python specifically this kata "Replace With Alphabet Position".  Although for the last 3 runs have returned really quickly, have you just done something to resolve?

**Comment by jhoffner** [comment]
No, it looks like some requests are really slow and some are not. This may be caused by "noisy neighbors"  - since we don't run on dedicated hardware. For a simple JS request I'm getting wall times (the time actually spent running the user code) between 100ms to 5 seconds.

**Comment by tehnicaorg** [comment]
There were timeouts with Python and JavaScript too, testing done in Europe.

**Comment by jhoffner** [comment]
Yeah turns out it was a huge surge in traffic, combined with a lot of people using Java, which is the language punishes the system resources the most. I scaled up and things are starting to get better.

**Comment by redscaresu** [comment]
does users java code get remotely compiled on your servers then any parsing errors returned back to the user?

**Comment by jhoffner** [comment]
It should. We recently rolled out some new updates so its possible there was a regression with one of the languages. Are you not getting anything back?

**Comment by redscaresu** [comment]
No its sorted thanks, funny enough ever since you replied it seems to have returned to normal.

hehe no wonder that is resource hungry, i noticed you are running on AWS...how come you dont stick the backend servers in an autoscale group?  Is it due to cost implications?

**Comment by tehnicaorg** [comment]
It's still too slow to be usable. I can now submit the code, but it fails to run with all tests.

For http://www.codewars.com/kata/5547cc7dcad755e480000004/train/java there are two tests visible.
Running the two tests (very basic, no loops) with a code that just returns pre-calculated values it takes 3.2 seconds to run.

```java
    public static List<long[]> removNb(long n) {
        if (n == 26) {
          List<long[]> res = new ArrayList<long[]>();
          res.add(new long[] {15, 21});
          res.add(new long[] {21, 15});
          return res;
        } else {
          return new ArrayList<long[]>();
        }
    }
```

Submitting the actual code always times out after 11 seconds.

**Comment by juliancoleman** [comment]
I'm having this issue currently for JavaScript and Python, the kata I've been having these issues on are

[Evaluate mathematical expression](https://www.codewars.com/kata/52a78825cdfc2cfc87000005)

[My smallest code interpreter](https://www.codewars.com/kata/526156943dfe7ce06200063e)

I've tried from over 10 computers, including my phone and iPad, on different networks. Ever since last night, these issues have been pretty spectacular, ranging from failed submissions, time-outs, 500 errors, as well as 522 errors, and a page that reads "Application Error". I've also received errors when logging in with Github. I hope I'm not the only one affected.
